skrub/_join_utils.py

```diff
@@ -218,3 +222,95 @@ def _get_new_name(suggested_name, forbidden_names):
         return suggested_name
     token = _utils.random_string()
     return f"{untagged_name}__skrub_{token}__"
+
+
+def left_join(left, right, left_on, right_on, rename_right_cols="{}"):
+    """Left join two dataframes of the same type.
+
+    The input dataframes type must agree: either both `left` and `right` need to be
+    :obj:`pandas.DataFrame`, :obj:`polars.DataFrame` or :obj:`polars.LazyFrame`.
```

Comment:
let's not mention LazyFrame for now; we don't need it in this private utility because we don't have support for lazy frames in the user-facing estimators